src/main/java/tennisclub/dao/BookingDaoImpl.java

```diff
@@ -0,0 +1,77 @@
+package tennisclub.dao;
+
+import org.springframework.stereotype.Repository;
+import tennisclub.entity.Booking;
+import tennisclub.entity.Court;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.time.LocalDateTime;
+import java.util.List;
+
+/**
+ * @author Miroslav Demek
+ */
+@Repository
+public class BookingDaoImpl implements BookingDao {
+
+    @PersistenceContext
+    private EntityManager em;
+
+    @Override
+    public void create(Booking booking) {
+        em.persist(booking);
+    }
+
+    @Override
+    public Booking findById(Long id) {
+        return em.find(Booking.class, id);
+    }
+
+    @Override
+    public List<Booking> findByCourt(Court court) {
+        return em.createQuery("select b from Booking b where b.court = :court", Booking.class)
+                .setParameter("court", court)
+                .getResultList();
+    }
+
+    @Override
+    public List<Booking> findByStartTime(LocalDateTime startTime) {
+        return em.createQuery("select b from Booking b where b.startTime = :startTime", Booking.class)
+                .setParameter("startTime", startTime)
+                .getResultList();
+    }
+
+    @Override
+    public List<Booking> findByEndTime(LocalDateTime endTime) {
+        return em.createQuery("select b from Booking b where b.endTime = :endTime", Booking.class)
+                .setParameter("endTime", endTime)
+                .getResultList();
+    }
+
+    @Override
+    public List<Booking> findByTimeInterval(LocalDateTime from, LocalDateTime to) {
+        return em.createQuery("select b from Booking b where b.endTime >= :from and b.startTime <= :to", Booking.class)
```

Comment:
This is a matter of opinion/use-case, but would strict inequalities (`>` and `<`) not make more sense? E.g. if an event **ends** at 16:00, I would not expect to see it in a list of events happening **from** 16:00 on.

Reply:
Hmmm, yes I think you are right. I will change it here and in EventDao as well.